File: server/QCLab/Models/ElectronicSignature.cs
using System;
using System.Collections.Generic;

namespace QCLab.Models;

public partial class ElectronicSignature
{
    public long Id { get; set; }

    public string EntityName { get; set; } = null!;

    public long EntityId { get; set; }

    public long SignerUserId { get; set; }

    public string SignatureMeaning { get; set; } = null!;

    public DateTime SigningTimestamp { get; set; }

    public string PayloadSha256 { get; set; } = null!;

    public string SignatureManifestText { get; set; } = null!;

    public string ClientIp { get; set; } = null!;

    public virtual User SignerUser { get; set; } = null!;
}

File: packages/client_react/src/types/electronicSignature.ts
export interface ElectronicSignatureDto {
  id: number;
  entityName: string;
  entityId: number;
  signerUserId: number;
  signerUserTag: string;
  signatureMeaning: string;
  signingTimestamp: string;
  payloadSha256: string;
  signatureManifestText: string;
  clientIp: string;
  isValid: boolean;
}

export interface ElectronicSignatureVerificationDto {
  isSigned: boolean;
  isValid: boolean;
  statusMessage: string;
  signatures: ElectronicSignatureDto[];
}
